go/runtime/history/indexer.go

```diff
@@ -230,6 +230,7 @@ func (bi *BlockIndexer) reindexTo(ctx context.Context, height int64) error {
 		lastHeight = lastRetainedHeight
 	}
 
+	batchSize := int64(config.GlobalConfig.Runtime.Indexer.BatchSize)
```

Comment:
We should avoid reading from the global state if possible. Therefore, you should accept this value in the constructor, either as required or optional parameter.